ZAPD/ZFile.cpp

```diff
@@ -793,11 +787,11 @@ string ZFile::ProcessDeclarations()
 		ProcessDeclarationText(item.second);
 	}
 
-	for (pair<int32_t, Declaration*> item : declarationKeysSorted)
+	for (pair<int32_t, Declaration*> item : declarations)
```

Comment:
Since you are changing this for performance reasons, you may want to use a reference in the iterator variable, so it doesn't perform a copy in each iteration.

```suggestion
	for (pair<int32_t, Declaration*>& item : declarations)
```

Reply:
I'm not sure that actually helps performance -- the loop uses both item.first and item.second, so the copy is needed anyway, and if it wasn't the compiler would be able to optimize it out. On the other hand, using a reference may require a read for each use of .first/.second, depending on the compiler's ability to do alias analysis. (The concrete impact on perf is probably very small, though.) IME references start being useful only when there are non-trivial copy constructors involved, like if you have a `pair<std::string, ...>`.

Reply:
My inclination (especially after simon's comment) is to leave it as is, but I don't feel strongly either way lol

Reply:
I would keep it as-is. Even if there's an optimization there, it's going to be negligible either way.